koku/sources/koku_http_client.py

```diff
@@ -115,6 +115,17 @@ def get_billing_source_for_provider(self, provider_type, billing_source):
         if provider_fn:
             return provider_fn(billing_source)
 
+    def _handle_bad_requests(self, response):
+        """Helper method to raise exceptions with error message string for Platform Sources."""
+        if response.status_code == 401 or response.status_code == 403:
+            raise KokuHTTPClientNonRecoverableError('Insufficient Permissions')
+        if response.status_code == 400:
```

Comment:
It might be good to have this catch any 4xx code:

```suggestion
        elif response.status_code >= 400 and response.status_code <= 499:
```

Reply:
So the 401 and 403 is returning an `Insufficient Permissions` string.
The actual error message string that we want to return is from a 400 (message from the koku provider method `cost_usage_source_is_reachable -> serializers.ValidationError(error_obj(key, message))`)

For the `create_provider` flow if it gets passed this check it is looking for any status code that's not the expected path
```
        self._handle_bad_requests(r)

        if r.status_code != 201:
            raise KokuHTTPClientNonRecoverableError(str(r.status_code))
```

Similar behavior on the `update_provider` method only there is also a `404` check

By the way, I'm working with docs/ux to review all of these strings and was planning on coming back and updating accordingly.

Reply:
That works for me! ^_^